shared/actions/favorite.js

```diff
@@ -172,23 +172,24 @@ export function favoriteList (): (dispatch: Dispatch) => void {
         dispatch(action)
         dispatch(badgeApp('newTLFs', !!(state.publicBadge || state.privateBadge)))
 
-        const newNotify = {public: state.publicBadge, private: state.privateBadge}
         const total = state.publicBadge + state.privateBadge
 
-        if (total && !_.isEqual(newNotify, previousNotify)) {
-          previousNotify = newNotify
+        if (total) {
+          const newNotifyState = [].concat(state.private.tlfs || [], state.public.tlfs || [])
+            .filter(t => t.meta === 'new').map(t => t.path)
 
-          let body
+          if (_.difference(newNotifyState, previousNotifyState).length) {
```

Comment:
I wan to see if there are new items not that it changed and became a subset
of itself. That's the original point of the ticket. The old code did is
equal

On Tuesday, July 12, 2016, Marco Munizaga <notifications@github.com> wrote:

> In shared/actions/favorite.js
> <https://github.com/keybase/client/pull/3409#discussion_r70513343>:
>
> >
> > -          let body
> > +          if (_.difference(newNotifyState, previousNotifyState).length) {
>
> then why difference instead of isEqual?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/client/pull/3409/files/a17bf13b46642d4ca6e534ad5b7c1f690a9c61fe#r70513343>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAfhU6v3IULOKc2Et4zZYYZuSn1JoLoCks5qU_XigaJpZM4JJkmV>
> .
>